Bugzilla/User.pm

```diff
@@ -726,12 +745,8 @@ sub nick {
     my $self = shift;
 
     return "" unless $self->id;
-
-    if (!defined $self->{nick}) {
-        $self->{nick} = (split(/@/, $self->login, 2))[0];
-    }
-
-    return $self->{nick};
+    return $self->{nickname} if $self->{nickname};
+    return $self->{nick} //= (split(/@/, $self->login, 2))[0];
```

Comment:
For right now, we use the old system.
We're not setting the nickname field to this because it doesn't make sense for ':dkl' to match dkl@mozilla.com if ':dkl' is not in the rename. Also it means the _migrate_nicknames() can complete fast, as only about 3k users use ircnicks like this.